### PR TITLE
Enables target discovery for Azure Web App step

### PR DESCRIPTION
**Pull request by geofflamrock**
As part of #project-dynamic-infrastructure we are adding support for discovering Azure Web App deployment targets. These targets are used by both the Azure App Service and Azure Web App steps.

This PR enables target discovery for Azure Web App steps by connecting the step with the Azure Web App deployment target type. The Sashimi PR https://github.com/OctopusDeploy/Sashimi/pull/132 needs to be completed and package references updated here before this can be shipped.

The discovery mechanism for the target is part of the AzureAppService repo and is being added in https://github.com/OctopusDeploy/Sashimi.AzureAppService/pull/24.

### CODE REVIEW

File: source/Sashimi/Sashimi.csproj [hunk 1]
```diff
@@ -23,9 +23,9 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Octopus.Configuration" Version="4.1.0" />
-    <PackageReference Include="Sashimi.Azure.Accounts" Version="14.1.3" />
-    <PackageReference Include="Sashimi.Azure.Common" Version="14.1.3" />
+    <PackageReference Include="Sashimi.Azure.Accounts" Version="14.2.0-andrewb-enh-targ0014" />
```
Comment (geofflamrock) [inline source/Sashimi/Sashimi.csproj:26]: References need to be updated once the relevant Sashimi PRs are complete.

File: source/Sashimi/Sashimi.csproj [hunk 2]
```diff
@@ -23,9 +23,9 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Octopus.Configuration" Version="4.1.0" />
-    <PackageReference Include="Sashimi.Azure.Accounts" Version="14.1.3" />
-    <PackageReference Include="Sashimi.Azure.Common" Version="14.1.3" />
+    <PackageReference Include="Sashimi.Azure.Accounts" Version="14.2.0" />
+    <PackageReference Include="Sashimi.Azure.Common" Version="14.2.0" />
     <PackageReference Include="Sashimi.AzureScripting" Version="14.1.0" />
```
Comment (gb-8) [inline source/Sashimi/Sashimi.csproj:28]: `Sashimi.AzureScripting` can be bumped to `14.2.0` too.

Comment (geofflamrock) [reply source/Sashimi/Sashimi.csproj]: Done, thanks

**Comment by gb-8** [comment]
`Calamari.Tests.Shared` reference in `Calamari.Tests.csproj` can also be updated to 14.2.0.
I don't think it make a real difference, but will one less line of noise in the build logs.

**Review by gb-8 — APPROVED** [review]

**Comment by geofflamrock** [comment]
> `Calamari.Tests.Shared` reference in `Calamari.Tests.csproj` can also be updated to 14.2.0. I don't think it make a real difference, but will one less line of noise in the build logs.

Ah yeah I didn't think to look in there as it wasn't complaining for me, updated now.

**Pull request merged**